### PR TITLE
fix: add serve

### PR DESCRIPTION
**Pull request by wtrocki**
Add very basic cli that will help us to run server using model file

### CODE REVIEW

File: bin/serve.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env node
+import { argv } from 'yargs';
+import { existsSync, readFileSync } from 'fs'
+import { TestxServer } from '../src';
+
+async function startServer(model: string) {
+    const server = new TestxServer({
```
Comment (darahayes) [inline bin/serve.ts:7]: Is it possible to assign a port instead of having one randomly generated?

Comment (wtrocki) [reply bin/serve.ts:7]: +100.

Comment (wtrocki) [reply bin/serve.ts]: Ok. This will mean refactoring of the testx as it was designed to bring random port :(

Comment (b1zzu) [reply bin/serve.ts:7]: The server start() method should be already abel to accpet a custom port, and if not is very easy to do it

File: bin/serve.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    // retrieve the server url
+    console.log(`Running on ${await server.httpUrl()}`);
+}
+
+
+if (argv.model) {
+    const modelLocation = argv.model as string;
+    if (existsSync(modelLocation)) {
+        const fileContent = readFileSync(modelLocation, { encoding: 'utf8' });
+        startServer(fileContent);
```
Comment (b1zzu) [inline bin/serve.ts:22]:
```suggestion
        startServer(fileContent).catch(e => throw e);
```
an alternative would be to log the error and stack and then exit with something different from 0, the only problem with the current way is that nodejs will give an error like `not handled exception in promise` which is not so nice

Comment (b1zzu) [reply bin/serve.ts:22]: the best approach will be to create a `main` function
```
async function main() {
  // everything here
}

main().catch(e => {throw e});
```

**Comment by b1zzu** [comment]
@wtrocki this is really great

**Comment by wtrocki** [comment]
This is just example what we can do. I do not intent to merge it here. 
I was thinking about separate package. More info here:

https://github.com/aerogear/graphback/issues/659

File: bin/serve.ts (hunk 1 above)
Comment (darahayes) [reply bin/serve.ts:7]: @wtrocki if you can take do something like `port = argv.port || '4001'` and then call it a day

File: bin/serve.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    await server.start();
+    // retrieve the server url
+    console.log(`Running on ${await server.httpUrl()}`);
+}
+
+
+if (argv.model) {
+    const modelLocation = argv.model as string;
+    if (existsSync(modelLocation)) {
+        const fileContent = readFileSync(modelLocation, { encoding: 'utf8' });
```
Comment (b1zzu) [inline bin/serve.ts:21]: What about moving the file read part inside Testx itself?
https://github.com/aerogear/graphql-testx/issues/97

File: bin/serve.ts (hunk 1 above)
Comment (b1zzu) [reply bin/serve.ts:7]: The TestxServer start is calling the GraphbackServer start function which accepts a fixed port, is just a matter of adding the argument to the TestxServer start and pass it through 
https://github.com/aerogear/graphql-testx/blob/db2a503e4a1ddf768d18b8eb888e75d6409a9e57/src/TestxServer.ts#L101
https://github.com/aerogear/graphql-testx/blob/db2a503e4a1ddf768d18b8eb888e75d6409a9e57/src/GraphbackServer.ts#L32

**Pull request closed without merging**